modis_tools/granule_handler.py

```diff
@@ -147,7 +148,11 @@ def download_from_urls(
 
     @classmethod
     def _get(
-        cls, url: HttpUrl, session: Session, stream: Optional[bool] = True
+        cls,
+        url: HttpUrl,
+        session: Session,
+        modis_session: ModisSession,
+        stream: Optional[bool] = True,
     ) -> Response:
```

Comment:
Now just imagine the pain adding a new param would have caused if we hadn't added the underscore to make it private. Conceptually we wouldn't have been confident an end-user wasn't dependent on `get()` in that exact format and we'd have to major version increment. 

Also noting that I love how your approach requires no changes whatsoever to the end-user access pattern. Not even additive code changes (e.g. adding a new method call)

Reply:
I definitely appreciated the foresight of making private methods private!